File: objc/fizzy/Fizzy.m
#import <Foundation/Foundation.h>
#import "Fizzy.h"

@implementation Fizzy
+ (NSString *) fizzy: (const NSInteger) n {
  const BOOL div3 = (n % 3 == 0);
  const BOOL div5 = (n % 5 == 0);

  if (div3 && div5) {
    return @"FizzBuzz";
  } else if (div3) {
    return @"Fizz";
  } else if (div5) {
    return @"Buzz";
  }

  return [NSString stringWithFormat:@"%ld", n];
}

int main() {
  for (int i = 1; i < 101; i++) {
    NSLog(@"%@", [Fizzy fizzy:i]);
  }

  return 0;
}
@end

File: clojure/fizzy/fizzy.clj
":";exec lein exec $0 ${1+"$@"}

(ns fizzy
  (:gen-class))

(defn fizzy [n]
  (let [x (mod n 3)
        y (mod n 5)]
    (cond (= [x y] [0 0]) "FizzBuzz"
          (= x 0) "Fizz"
          (= y 0) "Buzz"
          :else (format "%d" n))))

(defn -main [& args]
  (doall (map (comp println fizzy) (range 1 101))))

(when (.contains (first *command-line-args*) *source-path*)
  (apply -main (rest *command-line-args*)))
